### PR TITLE
mrs: alignment changes and cleanups

### PR DESCRIPTION
**Pull request by brooksdavis**
- Move the responsibility for alignment of small allocations into MRS
- remove OPTIONAL_QUARANTINING ifdef
- tidy JUST_INTERPOSE definition now that OPTIONAL_QUARANTINING isn't optional
- GC some unused bits

### CODE REVIEW

**Comment by bsdjhb** [comment]
I think BYPASS_QUARANTINE at least can also go?

**Comment by brooksdavis** [comment]
> I think BYPASS_QUARANTINE at least can also go?

I've removed it and rebased the PR

File: lib/libc/stdlib/malloc/mrs/mrs.c [hunk 1]
```diff
@@ -1333,7 +1345,8 @@ mrs_calloc(size_t number, size_t size)
 {
 #ifdef JUST_INTERPOSE
 	return (REAL(calloc)(number, size));
-#endif /* JUST_INTERPOSE */
+#else /* !JUST_INTERPOSE */
```
Comment (bsdjhb) [inline lib/libc/stdlib/malloc/mrs/mrs.c]: Part of the next commit?

File: lib/libc/stdlib/malloc/mrs/mrs.c [hunk 2]
```diff
@@ -1408,6 +1434,7 @@ mrs_calloc(size_t number, size_t size)
 
 	MRS_UTRACE(UTRACE_MRS_CALLOC, NULL, size, number, allocated_region);
 	return (allocated_region);
+#endif /* !JUST_INTERPOSE */
```
Comment (bsdjhb) [inline lib/libc/stdlib/malloc/mrs/mrs.c:1322]: Also part of next commit?

File: lib/libc/stdlib/malloc/mrs/mrs.c [hunk 3]
```diff
@@ -1002,10 +1000,8 @@ malloc_revoke(void)
 {
 	struct mrs_quarantine local_quarantine;
 
-#ifdef OPTIONAL_QUARANTINING
 	if (!quarantining)
 		return;
-#endif
```
Comment (bsdjhb) [inline lib/libc/stdlib/malloc/mrs/mrs.c:1010]: Maybe add a blank line here before the `MRS_UTRACE`?

File: lib/libc/stdlib/malloc/mrs/mrs.c [hunk 4]
```diff
@@ -1223,13 +1223,12 @@ fini(void)
 static void *
 mrs_malloc(size_t size)
 {
-#ifdef JUST_INTERPOSE
```
Comment (bsdjhb) [inline lib/libc/stdlib/malloc/mrs/mrs.c:1262]: I do wonder if we even still need `JUST_INTERPOSE`?  Presumably it was useful as a benchmark baseline, but probably not useful going forward?

File: lib/libc/stdlib/malloc/mrs/mrs.c [hunk 5]
```diff
@@ -320,8 +320,6 @@ static struct mrs_quarantine application_quarantine;
 static struct mrs_quarantine offload_quarantine;
 #endif /* OFFLOAD_QUARANTINE */
 
-static void *mrs_calloc_bootstrap(size_t number, size_t size);
```
Comment (bsdjhb) [inline lib/libc/stdlib/malloc/mrs/mrs.c:327]: For the commit log, s/can't the returned values/the returned values/

File: lib/libc/stdlib/malloc/mrs/mrs.c (hunk 4 above)
Comment (brooksdavis) [reply lib/libc/stdlib/malloc/mrs/mrs.c:1262]: I've removed it and actually realized we should remove all the `JUST_` ifdefs so I'll do that shorty.

**Pull request merged**